Description: Remove -march=native build flag.
 * This will cause gcc to use all instruction subsets
   available on the build machine, meaning that the program
   will not work on computers where they are not available.
 * Also the flag is not supported on many architectures
   (armhf, s390, s390x, powerpc, ia64, ...).
Author: Tobias Hansen <tobias.han@gmx.de>

Index: higan-094/Makefile
===================================================================
--- higan-094.orig/Makefile	2014-02-26 18:56:51.790230388 +0100
+++ higan-094/Makefile	2014-02-26 18:58:19.342229641 +0100
@@ -13,7 +13,7 @@
 # console := true
 
 # compiler
-flags   := -I. $(CFLAGS) $(CPPFLAGS)
+flags   := -I. -Dplatform=x -DPLATFORM_X=1
 link    := $(LDFLAGS)
 objects := libco
 
@@ -44,7 +44,6 @@
 else ifeq ($(platform),macosx)
   flags += -march=native
 else ifeq ($(platform),linux)
-  #flags += -march=native
   link += -Wl,-export-dynamic -lX11 -lXext -ldl
 else ifeq ($(platform),bsd)
   flags += -march=native
@@ -61,7 +60,7 @@
     $(if $(filter %.c,$<), \
       $(compiler) $(cflags) $(flags) $1 -c $< -o $@, \
       $(if $(filter %.cpp,$<), \
-        $(compiler) $(cppflags) $(flags) $1 -c $< -o $@ \
+        $(compiler) $(cppflags) $(flags) -std=c++0x -Dplatform=x -DPLATFORM_X=1 $1 -c $< -o $@ \
       ) \
     ) \
   )
Index: higan-094/phoenix/Makefile
===================================================================
--- higan-094.orig/phoenix/Makefile	2014-02-26 18:56:51.790230388 +0100
+++ higan-094/phoenix/Makefile	2014-02-26 18:56:51.786230388 +0100
@@ -13,12 +13,12 @@
   endif
 
   ifeq ($(phoenix),gtk)
-    phoenixflags = $(cppflags) $(flags) -DPHOENIX_GTK `pkg-config --cflags gtk+-2.0`
+    phoenixflags = $(cppflags) $(flags) -std=c++0x -Dplatform=x -DPLATFORM_X=1 -DPHOENIX_GTK `pkg-config --cflags gtk+-2.0`
     phoenixlink = `pkg-config --libs gtk+-2.0`
   endif
 
   ifeq ($(phoenix),qt)
-    phoenixflags = $(cppflags) $(flags) -DPHOENIX_QT `pkg-config --cflags QtCore QtGui`
+    phoenixflags = $(cppflags) $(flags) -std=c++0x -Dplatform=x -DPLATFORM_X=1 -DPHOENIX_QT `pkg-config --cflags QtCore QtGui`
     phoenixlink = `pkg-config --libs QtCore QtGui`
   endif
 endif
Index: higan-094/ruby/Makefile
===================================================================
--- higan-094.orig/ruby/Makefile	2014-02-26 18:56:51.790230388 +0100
+++ higan-094/ruby/Makefile	2014-02-26 19:01:09.666228186 +0100
@@ -1,7 +1,7 @@
 ifeq ($(platform),macosx)
   rubyflags = $(objcppflags) $(flags)
 else
-  rubyflags = $(cppflags) $(flags)
+  rubyflags = $(cppflags) $(flags) -std=c++0x -Dplatform=x -DPLATFORM_X=1
 endif
 
 rubyflags += $(foreach c,$(subst .,_,$(call strupper,$(ruby))),-D$c)
@@ -34,4 +34,5 @@
   rubylink += $(if $(findstring audio.openal,$(ruby)),-framework OpenAL)
 else
   rubylink += $(if $(findstring audio.openal,$(ruby)),-lopenal)
+  rubylink += -lm
 endif
Index: higan-094/nall/Makefile
===================================================================
--- higan-094.orig/nall/Makefile	2014-02-26 18:56:51.790230388 +0100
+++ higan-094/nall/Makefile	2014-02-26 19:02:00.366227754 +0100
@@ -49,8 +49,8 @@
     flags := -w -I/usr/local/include
   else
     compiler := g++
-    flags :=
-    link :=
+    flags := -std=c++0x
+    link := -lstdc++
   endif
 
   cflags := -x c -std=c99
Index: higan-094/ananke/Makefile
===================================================================
--- higan-094.orig/ananke/Makefile	2014-02-26 18:56:51.790230388 +0100
+++ higan-094/ananke/Makefile	2014-02-26 19:02:58.202227260 +0100
@@ -2,7 +2,7 @@
 include ../phoenix/Makefile
 
 path := /usr/local/lib
-flags := $(flags) $(CFLAGS) $(CPPFLAGS) -I..
+flags := $(flags) $(CFLAGS) $(CPPFLAGS) -I.. -std=c++0x -Dplatform=x -DPLATFORM_X=1
 
 all:
 	$(compiler) $(cppflags) $(flags) -fPIC -o obj/ananke.o -c ananke.cpp
@@ -13,7 +13,7 @@
 else ifeq ($(platform),macosx)
 	$(compiler) $(link) -shared -dynamiclib -undefined suppress -flat_namespace -o libananke.dylib obj/ananke.o
 else
-	$(compiler) $(link) -shared -Wl,-soname,libananke.so.1 -o libananke.so obj/ananke.o
+	$(compiler) $(link) -shared -Wl,-soname,libananke.so.1 -o libananke.so obj/ananke.o -lstdc++
 endif
 
 resource: force
